### DISCUSSION TITLE
Issue with Long Delay After DATA Command

### DISCUSSION THREAD

**Discussion by niels-przybilla** [postalserver/postal#3251] [Help with using Postal]
Hi Postal Community

I’m experiencing an issue with Postal where it seems to close the session prematurely during email delivery to an mail server that introduce a delay after the DATA command. Here’s what I’ve observed:

Testing with Telnet

./telnet mail.cp-ag.at 25
The session works as expected, and here’s a snippet:

220 mail.cp-ag.at Microsoft ESMTP MAIL Service ready
ehlo versand.sf-hosting.net  
250-mail.cp-ag.at Hello [116.203.41.0]
mail from: mihai@XXXX.net  
250 2.1.0 Sender OK  
rcpt to: xxx.xxxx@cp-group.com  
250 2.1.5 Recipient OK  
data  
354 Start mail input; end with <CRLF>.<CRLF>  
hello d  
.  
250 2.6.0 ... Queued mail for delivery

**Observation:**
After issuing the DATA command and inputting the email content, the server takes around 30 seconds to respond with a confirmation (e.g., 250 2.6.0 ... Queued mail for delivery).

I suspect that Postal does not wait long enough for this delay and closes the connection, causing a session.timeout error. This behavior likely prevents the email from being delivered successfully to such servers.

Timeout Configuration:
Is there a way to increase the timeout for the DATA response in Postal?

Are there alternative ways to handle these types of delays gracefully within Postal?

I’d appreciate any guidance or configuration tips to resolve this. Thank you in advance for your help! 😊

Best wishes

Niels

**Comment by willpower232** [comment]
It looks like you can configure that here

https://github.com/postalserver/postal/blob/fd3c7ccdf6dc4ee0a76c9523cbd735159e4b8000/doc/config/yaml.yml#L213-L217
